### PR TITLE
[DEM-515] ZI UHFLI instrument adapter.

### PR DESCRIPTION
**Pull request by CumulonimbusCalvus**
☁️

### CODE REVIEW

File: src/qilib/configuration_helper/adapters/common_instrument_adapter.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+import logging
+from abc import ABC
+
+from qilib.configuration_helper import InstrumentAdapter
+from qilib.utils import PythonJsonStructure
+
+
+class CommonInstrumentAdapter(InstrumentAdapter, ABC):
```
Comment (CumulonimbusCalvus) [inline src/qilib/configuration_helper/adapters/common_instrument_adapter.py]: It this additional ABC needed?

File: src/qilib/configuration_helper/adapters/common_instrument_adapter.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            config: The configuration with settings for the adapters instrument.
+
+        """
+        parameters = [parameter for parameter in config if hasattr(self._instrument.parameters[parameter], 'set')]
+        if any(config[parameter]['value'] is None for parameter in parameters):
+            error_message = 'Some parameter values of {0} are None and will not be set!'.format(self._instrument.name)
+            logging.warning(error_message)
+        for parameter in parameters:
+            if 'value' in config[parameter] and config[parameter]['value'] is not None:
+                self._instrument.set(parameter, config[parameter]['value'])
```
Comment (CumulonimbusCalvus) [inline src/qilib/configuration_helper/adapters/common_instrument_adapter.py]:
```
from qilib.configuration_helper import InstrumentAdapterFactory

adapter = InstrumentAdapterFactory.get_instrument_adapter('UhfliInstrumentAdapter', 'dev2338')
configuration = adapter.read()
adapter.apply(configuration)
```

This snippet gave me the following error at line 47: 

**TypeError: ('1.0 is not an int; Parameter: UhfliInstrumentAdapter_dev2338.demod1_harmonic', 'setting UhfliInstrumentAdapter_dev2338_demod1_harmonic to 1.0').**

File: src/tests/unittests/configuration_helper/adapters/test_uhfli_instrument_adapter.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        self.assertEqual(0.2, trigger_level)
+        config = adapter.read()
+
+        adapter.instrument.scope_trig_level(0.7)
+        trigger_level = adapter.instrument.scope_trig_level.raw_value
+        self.assertEqual(0.7, trigger_level)
+
+        adapter.apply(config)
+        trigger_level = adapter.instrument.scope_trig_level.raw_value
+        self.assertEqual(0.2, trigger_level)
```
Comment (CumulonimbusCalvus) [inline src/tests/unittests/configuration_helper/adapters/test_uhfli_instrument_adapter.py]:
```
adapter.instrument.close()
```

The unittest give a warning when not closing a QCoDeS instrument at the end of this test.

File: src/tests/unittests/configuration_helper/adapters/test_uhfli_instrument_adapter.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            self.assertIsInstance(adapter.instrument.scope_trig_hystabsolute.raw_value, np.int64)
+
+            config = adapter.read()
+            adapter.apply(config)
+
+            for parameter in adapter.instrument.parameters.values():
+                if hasattr(parameter, 'raw_value'):
+                    self.assertNotIsInstance(parameter.raw_value, np.int64)
+            logger_mock.warning.assert_called_with('Some parameter values of ZIUHFLIInstrumentAdapter_dev4142 are'
+                                                   ' None and will not be set!')
```
Comment (CumulonimbusCalvus) [inline src/tests/unittests/configuration_helper/adapters/test_uhfli_instrument_adapter.py]: It think we have the same closing warning here.

File: src/qilib/configuration_helper/adapters/common_instrument_adapter.py (hunk 1 above)
Comment (qSaevar) [reply src/qilib/configuration_helper/adapters/common_instrument_adapter.py]: yes, strange, but yes

File: src/tests/unittests/configuration_helper/adapters/test_uhfli_instrument_adapter.py (hunk 4 above)
Comment (qSaevar) [reply src/tests/unittests/configuration_helper/adapters/test_uhfli_instrument_adapter.py]: ✔️

File: src/tests/unittests/configuration_helper/adapters/test_uhfli_instrument_adapter.py (hunk 3 above)
Comment (qSaevar) [reply src/tests/unittests/configuration_helper/adapters/test_uhfli_instrument_adapter.py]: ✔️

File: src/qilib/configuration_helper/adapters/common_instrument_adapter.py (hunk 2 above)
Comment (qSaevar) [reply src/qilib/configuration_helper/adapters/common_instrument_adapter.py]: ahhh

Comment (qSaevar) [reply src/qilib/configuration_helper/adapters/common_instrument_adapter.py]: that is a qcodes bug

Comment (qSaevar) [reply src/qilib/configuration_helper/adapters/common_instrument_adapter.py]: here is the PR: https://github.com/QCoDeS/Qcodes/pull/1534

**Pull request merged**